### PR TITLE
Add info to slack migration docs

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR fills in information in the Slack migration docs to reflect findings in my experience with migrating from Slack to Mattermost

### CODE REVIEW

File: source/onboard/migrating-to-mattermost.rst
Comment (mickmister) [inline source/onboard/migrating-to-mattermost.rst:100]: I'm thinking the "Migrate from Slack" section should be at the top of the page, or this may need to just be its own page linked at the top of this page. I imagine Slack is likely where a customer is coming from.

Also, the sections in this page are quite random. There are several paragraphs of marketing-related talk about "Bespoke" before we get to the section about migrating from Slack. Also, both "Migrate Mattermost server" and "Migrate from Bitnami" are referring to migrating Mattermost servers to another Mattermost server, but this is not made clear.


image

File: source/onboard/migrating-to-mattermost.rst
Comment (mickmister) [inline source/onboard/migrating-to-mattermost.rst:175]: Highlighting my existing comment here

File: source/onboard/migrating-to-mattermost.rst
Comment (mickmister) [inline source/onboard/migrating-to-mattermost.rst:209]: Highlighting two existing comments here. The "Michael K Note" pieces will be removed. They are essentially TODOs right now

File: source/onboard/migrating-to-mattermost.rst
Comment (mickmister) [inline source/onboard/migrating-to-mattermost.rst:238]: TODO

File: source/onboard/migrating-to-mattermost.rst
Comment (cwarnermm) [reply source/onboard/migrating-to-mattermost.rst:100]: I agree with all of your feedback. This page hasn't been reviewed thoroughly in some time. Moving Slack up on the page makes good sense, and I'd expect bespoke systems to be near the bottom of the page after other named tools.

**Comment by cwarnermm** [comment]
Thank you, @mickmister! Let me know how best I can support you with this lift, and please feel free to reorder the content on the page based on your initial feedback.

**Comment by mickmister** [comment]
@cwarnermm I think doing a side-by-side comparison of the two versions of the rendered docs would be helpful. The markdown diff makes it hard to inspect the new changes, and seeing the new version without the context of the old will be difficult to know what the changes are

**Comment by cwarnermm** [comment]
@mickmister - I've reorganized the content on the page based on your excellent feedback in my commit. Also adjusted minor formatting to align with the documentation style guide.

File: source/onboard/migrating-to-mattermost.rst
Comment (mickmister) [reply source/onboard/migrating-to-mattermost.rst:17]: We have a dedicated page for the HipChat migration guide. I'm thinking we can put the slack migration docs in a separate page just the same. This will allow us to showcase that page easier from other places

image

Comment (cwarnermm) [reply source/onboard/migrating-to-mattermost.rst]: Thanks, @mickmister! I've split out the Slack content to its own page in my latest commit.

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:32]:
```suggestion
You will receive a zip file containing the following contents:
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:51]:
```suggestion
When you download your Slack export zip file, some data will be missing from the zip file:
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:52]:
```suggestion
- User emails
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:96]:
```suggestion
Now that you have a Slack export file with emails and attachments, let's convert this information into Mattermost's bulk import format using the import preparation tool ``mmetl``. Download the latest release of ``mmetl`` for your `OS and architecture <https://github.com/mattermost/mmetl/releases/>`__. Run ``mmetl help`` to learn more about using the tool.
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:122]:
```suggestion
Note that the migration is idempotent, meaning that you can run multiple imports that contain the same posts, and there will not be duplicated created posts in Mattermost. Each post is imported with the correct user/author and ``created_at`` value from your Slack instance. Threads are kept in tact with the import.
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:135]:
```suggestion
    ``export MM_FILESETTINGS_DIRECTORY=/absolute/path/to/mattermost/server/data``
```

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [inline source/onboard/migrate-from-slack.rst:133]: I think the "local development environment" scenario might be better suited to an FAQ or "Troubleshooting" section at the bottom.  I would think most readers will be admins looking to do this with official Mattermost releases.

@mickmister thoughts?

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [inline source/onboard/migrate-from-slack.rst:137]: I'm inclined to expect customers are working with an official release, and leave the `make` path to advanced users to figure out.

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [inline source/onboard/migrate-from-slack.rst:142]: Can we link to the `mmctl` doc here?

File: source/onboard/migrate-from-slack.rst
Comment (mickmister) [reply source/onboard/migrate-from-slack.rst:133]: @sadohert Sounds good to me. Also, maybe all internal testing should be done using S3/Minio and not the local file system setup. If you agree I'll add a piece about recommending external storage over local. There really shouldn't be any production deployments using the local filesystem for data storage anyway. (I've heard the horror stories of data loss)

File: source/onboard/migrate-from-slack.rst
Comment (mickmister) [reply source/onboard/migrate-from-slack.rst:137]: @sadohert Yes this is related to doing internal testing. I'll just assume we're using the production release for internal testing as well, and recommend that as such in the docs

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:104]:
```suggestion
The tool outputs a `.jsonl <https://jsonlines.org/examples>`__ file containing all of your users, channels, and posts. It also creates a ``data`` folder that contains all of your attachments.
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:120]:
```suggestion
For the second option, the server will save the import in its file store before running the import (e.g. AWS S3 if you are using that are your file store), so there will be time spent uploading/downloading the file in this case. Since this is really just a one-time thing to import this file, we recommend the first option of running it on the server itself if you have a large export zip file.
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:122]:
```suggestion
The migration is idempotent, meaning that you can run multiple imports that contain the same posts, and there will not be duplicated created posts in Mattermost. Each post is imported with the correct user/author and ``created_at`` value from your Slack instance. Threads are kept in tact with the import.
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [inline source/onboard/migrate-from-slack.rst:142]:
```suggestion
Ensure you have the Mattermost command line tool ``mmctl`` installed. This allows you to perform different tasks that communicate to Mattermost's API. You'll also want to `configure authentication </manage/mmctl-command-line-tool.html#mmctl-auth>`__ for the tool.
```

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [reply source/onboard/migrate-from-slack.rst:133]: @mickmister - I'll hold off approving this PR so that you can apply this update.

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [reply source/onboard/migrate-from-slack.rst:137]: Thanks, @mickmister!

**Comment by mickmister** [comment]
@fmartingr Requesting your review as a means to test the usage of the docs before merge. Here's the rendered version of the PR http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/6788/onboard/migrate-from-slack.html

**Comment by mickmister** [comment]
@cwarnermm Something we could add to the GitHub action that comments on these docs PRs - If the changes are limited to 10 files or less, we can have the bot provide links to those specific pages in its post, so the reviewer can go directly to the rendered versions of the changes and not have to navigate around the rendered docs site. It would also be useful to have part of the comment say "here's all of the files, and these files are the ones that changed in the last commit"

How might we expose this effort as a HW ticket? Maybe in the server repo?

File: source/onboard/migrate-from-slack.rst
Comment (fmartingr) [inline source/onboard/migrate-from-slack.rst:114]: We don't have a zip file at this point.

File: source/onboard/migrate-from-slack.rst
Comment (fmartingr) [inline source/onboard/migrate-from-slack.rst:102]: I had to use a team name without spaces. Tried `slack import` at first but it didn't work:

```
Data: map[error:Error during job execution. — BulkImport: Error importing channel. Team with name "slack import" could not be found., resource: Team id: name=slack import import_file:8jee3tjr1jd3xqi3mpbooe976r_mattermost-bulk-import.zip line_number:3]
```

After I changed it to `slack-import` it worked perfectly.

File: source/onboard/migrate-from-slack.rst
Comment (fmartingr) [inline source/onboard/migrate-from-slack.rst:110]: nit: Can we use other color than green in the important blocks? they don't seem to impact that much right now. Maybe in the new docs design?

File: source/onboard/migrate-from-slack.rst
Comment (fmartingr) [inline source/onboard/migrate-from-slack.rst:74]: Should we make a note that the `slack-advanced-exporter` is not a Mattermost _product_? (Because it isn't, right?)

File: source/onboard/migrate-from-slack.rst
Comment (fmartingr) [inline source/onboard/migrate-from-slack.rst:33]: Getting to the import/export button wasn't that easy. The correct path was:

```
Workspace menu > Tools & Settings > Workspace Settings > Import/Export data button > Export 
```

It wasn't trivial even coming back to that section, as I went to _Customize workspace_ rather than _Workspace settings_ after actually knowing where the button was.

Maybe we can put some screenshots here? I made some.

cc: @cwarnermm

**Review by fmartingr — CHANGES_REQUESTED** [review]
Hey @mickmister, nice work!

I got some time today to try this out: I had a slack workspace laying around (but was empty at this point) so I just put some messages and attachments in channels and tried this using `mmctl`.

I managed to import the data correctly (public channels only) but I have some comments in some sections that I would modify or are incorrect. Comments are below.

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [reply source/onboard/migrate-from-slack.rst:110]: @fmartingr - Loving your mindset here! Yes, in the new docs design, we have full control over not only the colour of callouts, but can customize the header text and icons beyond "Notes" or "Important".

I agree that "Note" blocks should be styled to appear different/distinct from "Important" and other callouts to draw attention appropriately.

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [reply source/onboard/migrate-from-slack.rst:74]: Transparency here will clarify ownership and build trust. Yes from me!

File: source/onboard/migrate-from-slack.rst
Comment (cwarnermm) [reply source/onboard/migrate-from-slack.rst:33]: Screenshots are always a good addition to docs. They help orient users in our product and validate that the user is in the correct place for the next step in a process. All for adding screenshots and I can help with this lift, once we've landed the PR.

File: source/onboard/migrate-from-slack.rst
Comment (mickmister) [reply source/onboard/migrate-from-slack.rst:74]: @fmartingr I would say that it is a Mattermost product, as it was written by a Mattermost employee at the time. What are your thoughts on getting this moved to the Mattermost GitHub org, as an alternative to noting that it is not a Mattermost product?

File: source/onboard/migrate-from-slack.rst
Comment (mickmister) [reply source/onboard/migrate-from-slack.rst:33]: I'm trying to get some screenshots but unfortunately the Slack webapp keeps crashing

> Maybe we can put some screenshots here? I made some.

@fmartingr Do you mean that you have screenshots to share?

**Comment by mickmister** [comment]
@cwarnermm FYI I created an issue to remove the marketing verbiage around bespoke https://github.com/mattermost/docs/issues/6816

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [inline source/onboard/migrate-from-slack.rst:173]:
```suggestion
If you choose the ``mmctl`` option above, you can use the ``mmctl import job show`` command to view any relevant errors that may have occurred. If you run into problems which the error message does not help to resolve, please try the second option of using the ``mattermost bulk import`` command. The ``mmctl`` import process does not give you any additional debugging information, even in the Mattermost server logs.
```

Comment (sadohert) [reply source/onboard/migrate-from-slack.rst:173]: I think the ordering changed, hence the suggested edit.

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [inline source/onboard/migrate-from-slack.rst:177]: What situation are we getting at here?  Are we kinda saying "Hey, if you are part of the user group being imported from Slack your Mattermost profile must have a matching username and email to the corresponding `User` line of the `jsonl` file.  You can manually edit the file to ensure make it match your Mattermost user"

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [inline source/onboard/migrate-from-slack.rst:189]: My first thought was to add something like:

We recommend administrators search the final import `jsonl` file for `user` lines for with `@example.com` in the email property to address and resolve the missing information prior to import.

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [inline source/onboard/migrate-from-slack.rst:191]: I think a link that directs admins to instructyions for migrating users authentication method would be helpful.  I'm not sure if all the instructions should go here, or simply link off to our tool.  I'm inclined to put a full example for migrating email/pw users to SAML.

**Review by sadohert — COMMENTED** [review]
@mickmister  fantastic job man!  I found this one to be a gnarly one to lay out... you have a nice flow through that incorporates all the related tools (except maybe one... see below).

One case I'm wondering about is when a customer is merging multiple workspaces, and what should our guidance be here... do we add considerations for this right on the same page, or link of somewhere else as an "Advanced Topic".  This is a gnarly topic itself, and introduces the need for the sync tool you built to help address user merge issues.

File: source/onboard/migrate-from-slack.rst
Comment (sadohert) [reply source/onboard/migrate-from-slack.rst:15]: We both kind ran into this around the same time I think.  I posted in [Developers: Server](https://github.com/mattermost/docs/pull/6788#discussion_r1411241200) about what I was having to do with running the command to get it to work, and Agniva shared https://mattermost.atlassian.net/browse/MM-55987 with me.

Ultimately this goes away when there's only the `mmctl` option.

> Are customers that are using the mounted filestore normally using a relative path for the local filestore?

I think its a mix.  Not sure I've seen a pattern.  I think often relative to the directory the MM process is started from... often `./data` I believe.

File: source/onboard/migrate-from-slack.rst
Comment (fmartingr) [reply source/onboard/migrate-from-slack.rst:33]: > I'm trying to get some screenshots but unfortunately the Slack webapp keeps crashing
> 
> > Maybe we can put some screenshots here? I made some.
> 
> @fmartingr Do you mean that you have screenshots to share?

Yes, I will share them with you in Community.

File: source/onboard/migrate-from-slack.rst
Comment (fmartingr) [reply source/onboard/migrate-from-slack.rst:74]: I'm ok with moving it to the org if we plan to actively support it, but I can't make that decision since I don't know which team would own it and if they have bandwidth for it.

**Review by fmartingr — APPROVED** [review]
👍 LGTM

**Comment by mickmister** [comment]
I reverted adding the screenshots, as that is specific to only exporting public channels from Slack. I think customers will mostly be doing the full export, which those screenshots don't help with, so I think they will more of a distraction than anything, especially because of how big the images are rendered on the docs page

**Review by sadohert — APPROVED** [review]
LGTM.  Nice one @mickmister !

**Pull request merged**
